WeakAuras/WeakAuras.lua

```diff
@@ -3049,49 +3072,74 @@ function WeakAuras.AddMany(table)
   end
 end
 
-local function validateUserConfig(data)
+local function validateUserConfig(options, config)
   local authorOptionKeys = {}
-  for index, option in ipairs(data.authorOptions) do
-    if option.key then
+  for index, option in ipairs(options) do
+    local optionClass = WeakAuras.author_option_classes[option.type]
+    if optionClass == "simple" then
       authorOptionKeys[option.key] = index
-      if data.config[option.key] == nil then
+      if config[option.key] == nil then
         if type(option.default) ~= "table" then
-          data.config[option.key] = option.default
+          config[option.key] = option.default
         else
-          data.config[option.key] = CopyTable(option.default)
+          config[option.key] = CopyTable(option.default)
         end
       end
+    elseif optionClass == "group" then
+      authorOptionKeys[option.key] = "group"
+      local subOptions = option.subOptions
+      if type(config[option.key]) ~= "table" then
+        config[option.key] = {}
+      end
+      local subConfig = config[option.key]
+      if option.groupType == "array" then
```

Comment:
There is quite a bit of black magic going on to make this validation work. To avoid that I would need to do another fairly large refactoring of custom options, which I may yet do.